### PR TITLE
Add low level rpc to sign taproot inputs 

### PR DESCRIPTION
**Pull request by altafan**
This adds a low level rpc `SignPsetWithSchnorrKey` that accepts and signs a partial tx.

This pset must contain the account(s) xpub as global field and each taproot input to be signed must specify a TapBip32Derivation field.

Internally, the app service retrieves the account index for a global xpub and contrcuts the derivation path required by the signing wallet to generate the correct private key.

Please @louisinger review.

### CODE REVIEW

**Comment by tiero** [comment]
I would add a test to serve as reference/example on how to use the API

File: pkg/wallet/single-sig/sign.go [hunk 1]
```diff
@@ -252,3 +324,98 @@ func (w *Wallet) signInput(
...
+	ptx *psetv2.Pset, inIndex int, derivationPath string,
+	sighashType txscript.SigHashType, genesisBlockHash *chainhash.Hash,
+) error {
+	signer, err := psetv2.NewSigner(ptx)
+	if err != nil {
+		return err
+	}
+	input := ptx.Inputs[inIndex]
+	if input.SigHashType == 0 {
+		if err := signer.AddInSighashType(inIndex, sighashType); err != nil {
```
Comment (louisinger) [inline pkg/wallet/single-sig/sign.go:338]: unnecessary if `sighashType` is SIGHASH_DEFAULT

File: pkg/wallet/single-sig/sign.go [hunk 2]
```diff
@@ -252,3 +324,98 @@ func (w *Wallet) signInput(
...
+			)
+			if err != nil {
+				return err
+			}
+
+			if err := signer.SignTaprootInputTapscriptSig(inIndex, *tapScriptSig); err != nil {
+				return err
+			}
+		}
+		// If there are no leaf hashes, sign as keypath
```
Comment (louisinger) [inline pkg/wallet/single-sig/sign.go:368]: it looks like key-path case is missing ?

File: pkg/wallet/single-sig/sign.go [hunk 3]
```diff
@@ -252,3 +324,98 @@ func (w *Wallet) signInput(
...
+		return nil, fmt.Errorf("failed to sign with schnorr: %s", err)
+	}
+	if !signature.Verify(hashForSignature[:], pubkey) {
+		return nil, fmt.Errorf("signature verification failed for input %d", inIndex)
+	}
+
+	return &psetv2.TapScriptSig{
+		PartialSig: psetv2.PartialSig{
+			PubKey:    schnorr.SerializePubKey(pubkey),
+			Signature: signature.Serialize(),
```
Comment (louisinger) [inline pkg/wallet/single-sig/sign.go:418]: schnorr signatures include the input sighash at the end of the slice **only if** it's not `SIGHASH_DEFAULT`

**Review by louisinger — CHANGES_REQUESTED** [review]

File: pkg/wallet/single-sig/sign.go (hunk 2 above)
Comment (altafan) [reply pkg/wallet/single-sig/sign.go:368]: key path should be line 369-383

**Review by louisinger — APPROVED** [review]

File: pkg/wallet/single-sig/sign.go [hunk 4]
```diff
@@ -252,3 +324,101 @@ func (w *Wallet) signInput(
...
+		return nil, err
+	}
+
+	input := ptx.Inputs[inIndex]
+	prevoutScripts := [][]byte{input.GetUtxo().Script}
+	prevoutAssets := [][]byte{input.ExplicitAsset}
+	value, _ := elementsutil.ValueToBytes(input.ExplicitValue)
+	prevoutValues := [][]byte{value}
+
+	hashForSignature := unsignedTx.HashForWitnessV1(
```
Comment (louisinger) [inline pkg/wallet/single-sig/sign.go:402]: `prevoutScripts`, `prevoutAssets` & `prevoutValues` must include **all the transaction inputs data** (not only the signed one)

Comment (louisinger) [reply pkg/wallet/single-sig/sign.go:402]: also, EplicitAsset & ExplicitValue can't be used. The witness utxo data must be used, see the changes there: https://github.com/altafan/ocean/pull/2

**Review by louisinger — APPROVED** [review]

**Pull request merged**